pallets/author-slot-filter/src/migration.rs

```diff
@@ -40,35 +41,40 @@ where
 	fn on_runtime_upgrade() -> Weight {
 		log::info!(target: "EligibleRatioToEligiblityCount", "starting migration");
 
-		if let Some(old_value) =
-			migration::get_storage_value::<Percent>(PALLET_NAME, ELIGIBLE_RATIO_ITEM_NAME, &[])
-		{
-			let total_authors = <T as Config>::PotentialAuthors::get().len();
-			let new_value: u32 = percent_of_num(old_value, total_authors as u32);
-			migration::put_storage_value(
-				PALLET_NAME,
-				ELIGIBLE_COUNT_ITEM_NAME,
-				&[],
-				NonZeroU32::new(new_value).unwrap_or(crate::pallet::DEFAULT_TOTAL_ELIGIBLE_AUTHORS),
-			);
-
-			let db_weights = T::DbWeight::get();
-			db_weights.write + db_weights.read
-		} else {
-			0
-		}
+		let old_value =
+			migration::get_storage_value::<Percent>(PALLET_NAME, ELIGIBLE_RATIO_ITEM_NAME, &[]);
+		let new_value = old_value
```

Comment:
Personally, I prefer to avoid having to specify defaults in multiple places, hence went with the functional approach, but if the team prefers the procedural style, I can change it to. 

```rust
               let new_value = if let Some(value) = old_value {
			let total_authors = <T as Config>::PotentialAuthors::get().len();
			let new_value = percent_of_num(value, total_authors as u32);
			NonZeroU32::new(new_value).unwrap_or(EligibilityValue::default())
		} else {
			EligibilityValue::default()
		};
```

Only disadvantage being a default specified in each branch. Alternatively we can remove the flatten (forgot about `and_then`), and specify default only once with:

```rust
               let new_value = old_value
			.and_then(|value| {
				let total_authors = <T as Config>::PotentialAuthors::get().len();
				let new_value = percent_of_num(value, total_authors as u32);
				NonZeroU32::new(new_value)
			})
			.unwrap_or(EligibilityValue::default());
```

thoughts?

Reply:
If you prefer  the functional approach, `and_then` is more readable here :)

Reply:
I do prefer it but only due to not having to define a default in each branch (which would then need to be tested to not diverge).
In general I'd like to go with whatever the majority of the team is comfortable with :)

Reply:
I have always thought that the succinctness of `if let some` is lost when there is an `else` branch attached to it...